### ISSUE TITLE
AI trading with each other

### ISSUE THREAD

**Issue by ByBurton** [Sh0wdown/UnofficialCrusaderPatch#41]
Would it be possible, to enable trading between allied ai's?
They do already ask for stuff, and send stuff to the player, so can that be used for trading with each other?
Just weirds me out, that this exists with the player, but not between computer opponents..

**Comment by PitchNeeded** [comment]
Good idea. It would depend on the AI's personality too though what it would send so Sheriff would still usually decline sending anything to another AI whereas Marshall would be much more generous.

**Comment by ByBurton** [comment]
Maybe....

**Comment by KingCaliph** [comment]
Sadly nobody ever solved this issue. I have played this game since I was a child and I always wished of this being a possibility.

**Comment by Lolasik011** [comment]
> Sadly nobody ever solved this issue. I have played this game since I was a child and I always wished of this being a possibility.

rekindle the topic

**Comment by LordHansCapon** [comment]
Well, it would be more like:
- **A** needs X goods
- Scan allies who have the goods
- If ally is an AI, check whether that AI needs that resource (this can range from simple to pretty complex)
- If ally was an AI and does not need the resource, substract resource from **B** and add it to **A**

By the way, I have played around chaning the rule regarding asking for Gold. There is a hardcoded deny when player asks for gold, but I wrote a simple check which looks like this:
- Player has to has less gold
- Player has to has at least 1/3 of AI gold
- AI must have over 10000 gold

But it was still OP...

**Comment by ByBurton** [comment]
@LordHansCapon It would also need to take into calculation the aic variable that handles trades (how much stuff to keep), so that the character of an ai is still there. Caliph wouldn't send anything but apples to his allies, as those are the only thing he keeps enough on stock to have some leftover after he is asked for 5 or 10 apples).

That said I am quite interested about your change regarding sending gold.

**Comment by Monsterfisch** [comment]
200% would be amazing to see an interaction between the AI characters apart from throwing troops at each others walls. it would, however, be fairly important to boost their economies across the board so that such a feat can even be feasible. The abbot doesn't amass such treasures in a short amount of time after all.

**Comment by ByBurton** [comment]
> 200% would be amazing to see an interaction between the AI characters apart from throwing troops at each others walls. it would, however, be fairly important to boost their economies across the board so that such a feat can even be feasible. The abbot doesn't amass such treasures in a short amount of time after all.

Not necessarily. You could tweak the gold thingy. But me for my part am against the ai sending gold around. Unless they are way above their investment threshold and not actively recruiting a siege force. 
(e.g. If(attacking and gold > threshold * 20) send (threshold / allied ai lords) gold to each allied ai lord. Output text if the player is in their team.
There should also be text output for normal goods being sent from ai to ai if the player is in their team.

**Comment by Monsterfisch** [comment]
I definitely agree that the ai shouldn't just send out gold all the time I am looking at it more in terms of help to recover from an enemy attack. So let's say caliph has been run down pretty badly and his economy is flat on its belly because marshall or the player has rekt his outer buildings etc. his team member ais could send him money/ resources to get back on his feet where otherwise the ai would just end up being dead in the water for a long time. 

tldr; trading would only occur if an ai is pretty trashed and can't get back on its feet in the short term.

This logic could then also be used to have allies send reinforcements to their Team members. implying a mechanic like that would be realistically achievable ofc

**Comment by ByBurton** [comment]
Although sending gold is rarily a good thing. Sending resources is often way more cost effective.

**Comment by Monsterfisch** [comment]
true but like presents in rl money is just more universally useful. Now it would be a lot better if the Ai knew how to use it properly but that's a different story :D and there are certainly more important tweaks to the ai that can be made to have it less dependent on a larger rework like this for sure. This idea is also more a jab at an attempt to ave the AI from its own incompetence anyhow. I am fairly certain that once the ai starts giving players actual trouble no one will want them to trade with one another anymore xD

**Comment by ByBurton** [comment]
But I can already see the ai sending small gifts of gold like 10 to 50, and the almost defeated ai using that for positive taxes instead of buying wood...
So sending wood or other resources and not implementing gold would be a smarter first step in my honest opinion.

**Comment by Monsterfisch** [comment]
yeah I would classify that under AI incompetence and would not like to see trading used to "fix" the ai in that regard but rather have "trading" be introduced after those kinks have been taken care of you know ai prioritization and all. ;)
